datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/MolecularTest.kt

```diff
@@ -14,7 +14,7 @@ interface MolecularTest {
     val hasSufficientPurity: Boolean
     val hasSufficientQuality: Boolean
 
-    fun testsGene(gene: String): Boolean
+    fun testsGene(gene: String, molecularTestTargets: List<MolecularTestTarget> = emptyList()): Boolean
```

Comment:
No big fan of defaults for parameters of classes, especially if they are not null.

In this case the default input is even sort of invalid -> it is impossible to test a gene but on zero targets. If we'd provide a default then `all values` would be better (if you only provide the tested gene, assume that gene has been tested in its entirety). 

Also, I think a `Set` would be better here than a `List`, there is no use-case for duplicates